oresat_configs/scripts/gen_kaitai.py

```diff
@@ -0,0 +1,226 @@
+"""Generate KaiTai for the beacon."""
+
+from argparse import ArgumentParser, Namespace
+from typing import Any, Optional, cast
+
+import canopen
+from yaml import dump
+
+from .. import Consts, OreSatConfig
+
+GEN_KAITAI = "generate beacon kaitai configuration"
+
+
+def build_parser(parser: ArgumentParser) -> ArgumentParser:
+    """Configures an ArgumentParser suitable for this script.
+
+    The given parser may be standalone or it may be used as a subcommand in another ArgumentParser.
+    """
+    parser.description = GEN_KAITAI
+    parser.add_argument(
+        "--oresat",
+        default=Consts.default().arg,
+        choices=[m.arg for m in Consts],
+        type=lambda x: x.lower().removeprefix("oresat"),
+        help="Oresat Mission. (Default: %(default)s)",
+    )
+    parser.add_argument(
+        "-d", "--dir-path", default=".", help="Output directory path. (Default: %(default)s)"
+    )
+    return parser
+
+
+def register_subparser(subparsers: Any) -> None:
+    """Registers an ArgumentParser as a subcommand of another parser.
+
+    Intended to be called by __main__.py for each script. Given the output of add_subparsers(),
+    (which I think is a subparser group, but is technically unspecified) this function should
+    create its own ArgumentParser via add_parser(). It must also set_default() the func argument
+    to designate the entry point into this script.
+    See https://docs.python.org/3/library/argparse.html#sub-commands, especially the end of that
+    section, for more.
+    """
+    parser = build_parser(subparsers.add_parser("xtce", help=GEN_KAITAI))
+    parser.set_defaults(func=GEN_KAITAI)
+
+
+CANOPEN_TO_KAITAI_DT = {
+    canopen.objectdictionary.BOOLEAN: "s",
+    canopen.objectdictionary.INTEGER8: "int8",
+    canopen.objectdictionary.INTEGER16: "i1",
+    canopen.objectdictionary.INTEGER32: "i2",
+    canopen.objectdictionary.INTEGER64: "i3",
+    canopen.objectdictionary.UNSIGNED8: "u1",
+    canopen.objectdictionary.UNSIGNED16: "u2",
+    canopen.objectdictionary.UNSIGNED32: "u3",
+    canopen.objectdictionary.UNSIGNED64: "u4",
+    canopen.objectdictionary.VISIBLE_STRING: "str",
+    canopen.objectdictionary.REAL32: "float",
+    canopen.objectdictionary.REAL64: "double",
```

Comment:
Looking at the kaitai user guide

- signed ints are `iX`
- unsigned ints are `uX`
- floats are `fX`
- bool is `bX`

where X is the number of bytes

Reply:
Yeah that's right.

Most of the vars that matter reflect that (all the unsigned vars), only ones I needed to check first then update during manual regression was the bitfields because I needed to confirm what we're using first.

Last I checked, bitfields are just 1 byte- is this still true?

Reply:
All bitfields currently in the beacon are, but it not guaranteed to stay that way

You can check if an object is a bitfield by check if the object's `bit_definitions` is non-empty

Reply:
updated